src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -321,23 +321,27 @@ public bool TryReadTo(out ReadOnlySequence<T> sequence, T delimiter, T delimiter
         /// <param name="delimiters">The delimiters to look for.</param>
         /// <param name="advancePastDelimiter">True to move past the first found instance of any of the given <paramref name="delimiters"/>.</param>
         /// <returns>True if any of the the <paramref name="delimiters"/> were found.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public bool TryReadToAny(out ReadOnlySpan<T> span, ReadOnlySpan<T> delimiters, bool advancePastDelimiter = true)
         {
             ReadOnlySpan<T> remaining = UnreadSpan;
-            int index = remaining.IndexOfAny(delimiters);
+            int index = delimiters.Length == 2
```

Comment:
Why optimize specifically for 2 only?

Reply:
Because it didn't measure any improvements past that.